onadata/apps/messaging/viewsets.py

```diff
@@ -31,3 +35,35 @@ class MessagingViewSet(mixins.CreateModelMixin, mixins.ListModelMixin,
     filter_backends = (TargetTypeFilterBackend, TargetIDFilterBackend,
                        UserFilterBackend, DjangoFilterBackend)
     filterset_class = ActionFilterSet
+    pagination_class = StandardPageNumberPagination
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+        no_of_records = queryset.count()
+        retrieval_threshold = getattr(
+            settings, "MESSAGE_RETRIEVAL_THRESHOLD", 10000)
+        pagination_keys = [self.paginator.page_query_param,
+                           self.paginator.page_size_query_param]
+        query_param_keys = self.request.query_params
+        should_paginate = any(
+            [k in query_param_keys for k in pagination_keys]) or \
+            no_of_records > retrieval_threshold
+        current_page_size = self.request.query_params.get(
+            self.paginator.page_size_query_param) or retrieval_threshold
+        current_page = self.request.query_params.get(
+            self.paginator.page_query_param) or 1
+        headers = generate_pagination_headers(
+            self.request, no_of_records, current_page_size, current_page
+        )
```

Comment:
It seems this is run everytime we hit this endpoint.
I see above in line 48, we define if we `should_paginate` flag.
I'm curious if there is need to generate the pagination headers if `should_paginate` is False?

Reply:
No, it shouldn't. I'll update the code